### PR TITLE
feat(route/apnews): Add support for API

### PR DESCRIPTION
**Pull request by dzx-dzx**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/apnews/api?limit=1
/apnews/api/general-news?limit=1
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [x] Anti-bot or rate limit / 反爬/频率限制
  - [x] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

**Comment by dzx-dzx** [comment]
@TongRL Would you review on either of the PRs for the apnews route?

File: lib/routes/apnews/api.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import { parseDate } from '@/utils/parse-date';
+
+export const route: Route = {
+    path: '/api/:tags?',
+    categories: ['traditional-media', 'popular'],
+    example: '/apnews/api/business',
+    view: ViewType.Articles,
+    parameters: {
+        tags: {
+            description: 'See https://github.com/kovidgoyal/calibre/blob/81666219718b5f57d56b149a7ac017cc2a76b931/recipes/ap.recipe#L43-L46',
```
Comment (TonyRL) [inline lib/routes/apnews/api.ts:14]: Please do provide a description instead of an external link

File: lib/routes/apnews/api.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['apnews.com/'],
+        },
+    ],
+    name: 'News',
+    maintainers: ['zoenglinghou', 'mjysci', 'TonyRL'],
```
Comment (TonyRL) [inline lib/routes/apnews/api.ts:32]: Wrong maintainers

File: lib/routes/apnews/api.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import { Route, ViewType } from '@/types';
+import { fetchArticle } from './utils';
+import ofetch from '@/utils/ofetch';
+import timezone from '@/utils/timezone';
+import { parseDate } from '@/utils/parse-date';
+
+export const route: Route = {
+    path: '/api/:tags?',
+    categories: ['traditional-media', 'popular'],
```
Comment (TonyRL) [inline lib/routes/apnews/api.ts:9]: A purposing route can't be popular

File: lib/routes/apnews/api.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+async function handler(ctx) {
+    const { tags = 'apf-topnews' } = ctx.req.param();
+    const apiRootUrl = 'https://afs-prod.appspot.com/api/v2/feed/tag';
+    const url = `${apiRootUrl}?tags=${tags}`;
+    const res = await ofetch(url);
+
+    const list = res.cards
+        .map((e) => ({
+            title: e.contents[0].headline,
+            link: e.contents[0].localLinkUrl,
```
Comment (TonyRL) [inline lib/routes/apnews/api.ts:45]: Not every `cards` has `contents[0]`

<details>
<summary>Details</summary>

image


</details>

Comment (dzx-dzx) [reply lib/routes/apnews/api.ts:45]: What exactly is the content of the whole response?

**Pull request merged**